api-presigned.go

```diff
@@ -30,7 +30,7 @@ import (
 
 // presignURL - Returns a presigned URL for an input 'method'.
 // Expires maximum is 7days - ie. 604800 and minimum is 1.
-func (c Client) presignURL(ctx context.Context, method string, bucketName string, objectName string, expires time.Duration, reqParams url.Values) (u *url.URL, err error) {
+func (c Client) presignURL(ctx context.Context, method string, bucketName string, objectName string, expires time.Duration, reqParams url.Values, extraHeaders http.Header) (u *url.URL, err error) {
```

Comment:
Am I missing something here? Providing support at the internal function level for `presignURL()` is not good enough, because the end user has only access to the following 3 apis as documented in [MinIO Go Client API Reference](https://docs.minio.io/docs/golang-client-api-reference.html) :
- PresignedGetObject
- PresignedPutObject
- PresignedHeadObject

When end-user calls one of the above apis, s/he would expect to pass the extra headers to the api.

As the continuation of the above comment, we also need to modify the corresponiding scripts; `examples/s3/presignedgetobject.go`, `examples/s3/presignedheadobject.go`, and `examples/s3/presignedputobject.go`.

Reply:
@ebozduman

> Providing support at the internal function level for presignURL() is not good enough, because the end user has only access to the following 3 apis as documented in MinIO Go Client API Reference :

We also export `Presign()` (see [here](https://pkg.go.dev/github.com/minio/minio-go/v7#Client.Presign)) though we don't officially document it on our docs.min.io site. Since this change is expected to be useful only to some users, I've added this feature there. This helps prevent breaking backwards compatibility - however I am open to changing this if we think this can be more universally useful. CC: @harshavardhana

Reply:
OK. 
I'll run some tests and get back to you.

Reply:
I tested with my script and tried GET and PUT methods. They both passed successfully.
I also used Swagger tool to add the header on the browser and it was successful.
LGTM